docs/spec_draft.md

```diff
@@ -0,0 +1,124 @@
+# StableHLO Specification Draft
+
+## Types
+
+Following are the supported element types in StableHLO:
+
+  * **Integer types**
+
+    * Signed integer with two’s complement representation. Referred to in the
+    document as `si<N>`, where the bit-width N ∊ {4, 8, 16, 32, 64}
+    * Unsigned integer referred to in the document as `ui<N>`, where the
+    bit-width N ∊ {4, 8, 16, 32, 64}
+  * **Boolean types** referred to in the document as `pred`
+  * **Floating point types**
+    * Single precision `f32`, double precision `f64` and half precision `f16`
+    floating points complying with [IEEE 754
+    format](https://ieeexplore.ieee.org/document/8766229).
+    * Bfloat16 `bf16` floating point complying with [Brain Floating Point Format](https://cloud.google.com/blog/products/ai-machine-learning/bfloat16-the-secret-to-high-performance-on-cloud-tpus).
+    Provides the same number of exponent bits as `f32`, so that it matches its
+    dynamic range, but with greatly reduced precision.
+ * **Complex types** represents a pair of floating point types. Supported ones
+ are `c64` (represents paired `f32`) and `c128` (represents paired `f64`).
+
+StableHLO supports a Tensor type `tensor`, to model the type of a n-dimensional
+array, represented in IR as `tensor<SxE>` such that
```

Comment:
How about "represented in the opset", given that the current plan of record is for StableHLO to not be an IR?

Reply:
done.